pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1809,9 +1811,25 @@ public TopicStatsImpl getStats(boolean getPreciseBacklog, boolean subscriptionBa
         stats.lastOffloadLedgerId = ledger.getLastOffloadedLedgerId();
         stats.lastOffloadSuccessTimeStamp = ledger.getLastOffloadedSuccessTimestamp();
         stats.lastOffloadFailureTimeStamp = ledger.getLastOffloadedFailureTimestamp();
+        stats.compactedTopicSize = getCompactorMXBean().map(stat -> stat.getCompactedTopicSize(topic)).orElse(0D);
+        stats.compactedTopicMsgCount = getCompactorMXBean().map(stat ->
+                stat.getCompactedTopicMsgCount(topic)).orElse(0L);
+        stats.compactedTopicRemovedEventCount = getCompactorMXBean().map(stat ->
+                stat.getCompactedTopicRemovedEventCount(topic)).orElse(0L);
+        log.info("stats : {}", stats);
         return stats;
     }
 
+    private Optional<CompactorMXBean> getCompactorMXBean() {
+        Compactor compactor = null;
+        try {
+            compactor = brokerService.pulsar().getCompactor();
```

Comment:
It looks like the call to `getCompactor` initializes a `TwoPhaseCompactor` class, if it hasn't already been initialized. Currently, the class is only initialized for a broker when a topic hosted by the broker has compaction turned on. Wouldn't we want to avoid initializing this class, if possible? One way to do this would be to add a method to the `PulsarService` class that optionally gets the compactor or returns an empty option if it hasn't been initialized.

Reply:
@Technoboy- why did this get resolved? It looks like the `getCompactor` method is still being called, so my concerns are still present.

Reply:
Sorry, mistake for this, I will focus on this.

Reply:
Hi @michaeljmarshall , I have added a new method to avoid initializing the compactor. Could you help to review ?